core/pta/device.c

```diff
@@ -21,18 +21,23 @@
 #define PTA_NAME "device.pta"
 
 static void add_ta(uint32_t flags, const TEE_UUID *uuid, uint8_t *buf,
-		   uint32_t blen, uint32_t *pos)
+		   uint32_t blen, uint32_t *pos, bool supplicant)
```

Comment:
How about supply the required flags to this function directly instead? Should result in easier logic.

Reply:
@jenswi-linaro do you mean get_devices(nParamTypes, pParams, nCommandID) then then pass this nCommandID to add_ta() ?

Reply:
No, rather
``` C
case PTA_CMD_GET_DEVICES:
        return get_devices(nParamTypes, pParams, TA_FLAG_DEVICE_ENUM);
case PTA_CMD_GET_DEVICES_SUPP:
        return get_devices(nParamTypes, pParams, TA_FLAG_DEVICE_ENUM_SUPP);

Reply:
yea, that also looks nice.